### ISSUE TITLE
TweedieDevianceScore producing nan results when setting power=1.0

### ISSUE THREAD

**Issue by EvanZ** [PyTorchLightning/metrics#846]
### To Reproduce

```>>> td=TweedieDevianceScore(power=1.0)
>>> targets=torch.tensor([0,1,0,1])
>>> preds=torch.tensor([0.1,0.1,0.1,0.1])
>>> td(preds,targets)
tensor(nan)
>>> td=TweedieDevianceScore(power=1.0001)
>>> td(preds,targets)
tensor(1.5038)```

I assume this is a bug since the documentation says targets >= 0 (obviously Poisson data will often be 0).

**Comment by SkafteNicki** [comment]
Hi @EvanZ, thanks for reporting this. This is indeed a bug and it happens due to a unsafe log operation in the metric calculation. Will be fixed in PR #847.

**Comment by EvanZ** [comment]
Oh that's great to hear. Thanks!

**Comment by EvanZ** [comment]
Don't know if anyone will ever read this but here goes. Thanks for fixing the bug. It definitely works now in terms of calculating the loss. However, a secondary problem has arisen, which maybe is just unfixable. I'm trying to use the deviance score as a loss function. There exists a poisson loss function in PyTorch, which I'm currently using, but it would be convenient to use TweedieDevianceScore because that way I could model Poisson, Gamma, and mixed Poisson-Gamma distributions with the same code, only changing one parameter. 

Anyway, to the point. I am able to use TweedieDevianceScore as a loss function when power is set to 2.0 (Gamma) or in the range 1-2 (Poisson-Gamma). But now that I am trying to set power=1.0, I am getting `nan` values in the second batch of training. I checked and there are no `nan` values in the first batch and a finite loss is calculated. But for some reason in the second batch of training, `nan` values pop up. 

My guess is that it is an issue with the derivative when `power=1.0` and somehow even though the loss can be calculated now (thanks to this bug fix), the derivative of the loss is blowing up. If this is the case, I will probably just give up and use the built-in Poisson loss. But I thought I'd mention it here for posterity and in case anyone might have a clue. Maybe there's something else going on...

**Comment by SkafteNicki** [comment]
Hi @EvanZ,
Thanks for reporting this back. To easier debug this (and maybe solve) would it be possible for you to provide an example input that result in Nan?

**Comment by EvanZ** [comment]
@SkafteNicki appreciate your quick reply. The problem as I'm trying to explain (maybe not well) is that it's not the `TweedieDevianceScore` that is directly giving me `nan` values. **You guys definitely fixed that issue**. This "new" issue is occurring at some point after the loss for the first batch is calculated and backprop step (I assume this is when the `nan` values are being generated). But it definitely has *something* to do with setting `power=1.0` in `TweedieDevianceScore` because it's not happening with other values (1.01, 1.5, 2.0, etc). I wish I could paste my entire model and dataset, but I don't think that's possible (my employers wouldn't like it!).

As I said, I think this may be very difficult to figure out. But my guess is that it has something to do with autograd . Maybe `TweedieDevianceScore` is not differentiable when `power=1.0`?
